client-spark/spark-3-4/src/main/scala/org/apache/spark/shuffle/celeborn/CelebornShuffleReader.scala

```diff
@@ -370,7 +376,22 @@ class CelebornShuffleReader[K, C](
     }
   }
 
-  private def handleFetchExceptions(
+  @VisibleForTesting
+  def checkAndReportFetchFailureForUpdateFileGroupFailure(
+      celebornShuffleId: Int,
+      ce: Throwable): Unit = {
+    if (ce.getCause != null &&
+      (ce.getCause.isInstanceOf[InterruptedException] || ce.getCause.isInstanceOf[
```

Comment:
Here for the `InterruptedException`, even https://github.com/apache/celeborn/pull/2921 can handle it.

But this PR is more efficient, it do not need to talk with driver and prevent the stage-run for `InterruptedException` directly.